drivers/modem/gsm_ppp.c

```diff
@@ -821,7 +821,7 @@ static void gsm_finalize_connection(struct k_work *work)
 		/* Read connection quality (RSSI) before PPP carrier is ON */
 		query_rssi_nolock(gsm);
 
-		if (!((gsm->minfo.mdm_rssi > 0) && (gsm->minfo.mdm_rssi != GSM_RSSI_INVALID) &&
+		if (!((gsm->minfo.mdm_rssi) && (gsm->minfo.mdm_rssi != GSM_RSSI_INVALID) &&
```

Comment:
That's on me, the intention was to explicitly compare with a number to produce a boolean, I think it should be `< 0` instead?

Reply:
Actually I think this condition should be removed altogether. 
Setting `< 0` won't have effect as last condition is `(gsm->minfo.mdm_rssi < GSM_RSSI_MAXVAL)`.

`GSM_RSSI_MAXVAL` is defined conditionally based on `CONFIG_MODEM_GSM_ENABLE_CESQ_RSSI`.

https://github.com/zephyrproject-rtos/zephyr/blob/a94263784bc3b5f912a25ba231f7cabcfdd614f3/drivers/modem/gsm_ppp.c#L47-L51

So in case of `CONFIG_MODEM_GSM_ENABLE_CESQ_RSSI` not being set, last condition evaluates to `(gsm->minfo.mdm_rssi < -51)` which is stronger than `< 0`. Note that this should be `<= -51` as it's a valid value per specs.
When `CONFIG_MODEM_GSM_ENABLE_CESQ_RSSI` is set, then last condition matches first
`(gsm->minfo.mdm_rssi < 0)`.

Reply:
I think we need to study this further, meanwhile I think it is better to have `(gsm->minfo.mdm_rssi != 0)` instead so it is same as previous implementation but tested explicitly against 0